File: OkCompu/backend/backend-okcompu/src/main/java/com/okcompu/ecommerce/backendokcompu/model/Cliente.java
package com.okcompu.ecommerce.backendokcompu.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

@Entity
@Table(name="clientes")
@Data
@EqualsAndHashCode(onlyExplicitlyIncluded = true)  //, callSuper = true
public class Cliente  { //extends Base

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long idCliente;

    @Column(nullable = false)
    private String nombres;

    private String apellidos;

    @ManyToOne
    @JoinColumn(name = "id_tipodocumento", nullable = false, foreignKey = @ForeignKey(name = "FK_CLIENTE_TIPODOCUMENTO"))
    private TipoDocumento tipoDocumento;

    private String nroDocumento;

    @Column(unique = true)
    private String email;

    private String foto;

    private Byte estado;

    @OneToMany(mappedBy = "cliente", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Telefono> telefonos;

    @OneToMany(mappedBy = "cliente", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Domicilio> domicilios;
}

File: OkCompu/frontend/frontend-okcompu/src/app/model/cliente.ts
import { Domicilio } from "./domicilio";
import { Telefono } from "./telefono";
import { TipoDocumento } from "./tipo-documento";

// export interface Cliente {
export class Cliente {
    idCliente?: number;
    nombres: string;
    apellidos: string;
    tipoDocumento: TipoDocumento;
    nroDocumento: string;
    email: string;
    foto?: string;
    estado: number;
    telefonos: Telefono[];
    domicilios: Domicilio[];
}